### PR TITLE
Update requirements etc to msprime 1

### PR DESCRIPTION
**Pull request by hyanwong**
And point tskit docs to tskit.dev.

I haven't updated workflows. I guess this means grepping for `msprime.simulate` in the tuts, and replacing them with sim_ancestry/sim_mutations?

### CODE REVIEW

**Review by jeromekelleher — CHANGES_REQUESTED** [review]
LGTM, but we need to update the workflows file to remove the explicit msprime install. Might as well do the check for ``simulate`` as you're making the change.

**Comment by hyanwong** [comment]
Yep, just checking that was what was meant by "updating workflows".

**Comment by hyanwong** [comment]
Workflow updated and the 2 tuts that used `msprime.simulate` have been changed to sim_ancestry. Before merging, probably worth correcting the "noty convincing" typo in the bottlenecks tut I mentioned in the documentation slack channel.

File: introgression.md [hunk 1]
```diff
@@ -90,21 +87,21 @@ def get_migrating_tracts(ts):
...
+        if ts.population(migration.dest).metadata['name'] == 'Neanderthal':
             migrating_tracts.append((migration.left, migration.right))
     return np.array(migrating_tracts) 
 
 def get_coalescing_tracts(ts):
     coalescing_tracts = []
     tract_left = None
-    for tree in ts.trees():    
+    for tree in ts.trees():
         # 1 is the Eurasian sample and 2 is the Neanderthal
```
Comment (jeromekelleher) [inline introgression.md:98]: I think it'd be better to stick with the integer population IDs here - it's quite a lot less efficient to decode the metadata etc, than to compare integers, and we're already hard-coding in 1 and 2 as the Eurasian IDs etc. You could do

```
neanderthal_id = demography["Neanderthal"].id
...
if mrca_pop == neaderthal_id
```

**Review by jeromekelleher — COMMENTED** [review]
LGTM, one minor comment.

File: introgression.md (hunk 1 above)
Comment (hyanwong) [reply introgression.md:98]: I didn't know about `demography["Neanderthal"]`. Good call.

Comment (hyanwong) [reply introgression.md:98]: Actually, we don't really have access to the demography object within those functions (or we shouldn't, anyway, unless we treat demography as global, or return it from `run_simulation`). So probably better to put

```
neanderthal_id = [p.id for p in ts.populations() if p.metadata['name']=='Neanderthal'][0]
```

at the top of the function?

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**